### PR TITLE
Fixed VeinMiner not dropping xp for blocks mined with VeinMiner

### LINKED ISSUE #33: VeinMiner not dropping XP for mined ores.
### Description
XP is not dropped from any blocks mined using the vein miner enchant ability.

### Steps to reproduce
Just veinmine ores that drop xp and watch it not drop xp.

### PR DESCRIPTION
**Pull request by NRProjects**
Fixes #33 

## Description
Fixes the veinminer xp bug by calculating the xp and dropping it manually since ```breakNaturally()``` and neither does ```dropItemNaturally()```
```breakBlockNaturally``` also doesn't calculate for fortune

Added JavaDocs

### CODE REVIEW

**Comment by Teelair** [comment]
player#breakBlock(block)

**Comment by Teelair** [comment]
This could be solved with `player.breakBlock(block)`. Also, you have a bunch of logging in here.

**Comment by NRProjects** [comment]
> This could be solved with `player.breakBlock(block)`. Also, you have a bunch of logging in here. 

Forget to mention in the PR, but that doesn't drop XP, or calculate for fortune, or both. It was the first thing I tried and forgot to document why it didn't work

**Comment by paulpugh000** [comment]
Tested this method yesterday. It does work, it does drop the correct amount of XP. 
Here's my test I did:
Started off with this pickaxe, Fortune 3 Unbreaking 3, Vein Miner with the player.breakBlock(block) method used.

I used this 3x3x9, 54 blocks of Nether Quartz to see if the changes worked. Quartz drops a lot of XP, so I used it as the medium.

Here's the drops I got from using Vein Miner (crouching and mining)

The method does work, and it solves the other pitfalls that weren't addressed here, such as not taking into account Fortune and Silk Touch. But because player.BreakBlock(block) does call a BlockBreakEvent, you need to make sure you aren't doing this recursively or you might get stack overflow. I think this might have happened in the cases you tested, but even when not adding the guard rails for recursion, it still drops the proper amount of XP, just not the items. (Until the server crashes)

**Comment by NRProjects** [comment]
Yeah those were basically my findings. This would have to happen recursively since we would be calling it for every block stored in the set. Probably could solve the overflow issue by lowering the max amount of veinmineable blocks per veinmine but the way implemented in the PR solves the issue without limiting the user experience

**Review by Teelair — CHANGES_REQUESTED** [review]

File: src/main/java/plugins/nate/smp/listeners/enchantments/VeinMinerListener.java [hunk 1]
```diff
@@ -50,23 +36,80 @@ public class VeinMinerListener implements Listener {
...
+     *     <li><strong>Sequential Block Breaking:</strong> The vein mining process breaks all related blocks in a
+     *     sequence, rather than recursively. This sequential approach avoids triggering new instances of
+     *     BlockBreakEvent for the same player during vein mining.</li>
+     *     <li><strong>Guard Conditions:</strong> Several conditions are checked (e.g., player game mode, sneaking
+     *     status, tool enchantment) before initiating the vein mining process, ensuring that it only occurs under
+     *     appropriate circumstances.</li>
+     * </ul>
+     *
+     * @param event The block break event from the server.
+     */
```
[the remark this answers is not in the document]
Comment (NRProjects) [reply src/main/java/plugins/nate/smp/listeners/enchantments/VeinMinerListener.java:62]: I didnt star the comment train and I ain't got time to write javadocs

**Review by Teelair — APPROVED** [review]
LGTM! :shipit:

**Pull request merged**